Build failure: monero-cli

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `monero-cli` on current master (7dcb1a29b6da3524f4ae5499f5c348dde40a711a)

## Build log

<details>

<summary>Build Log</summary>

```
[...]
monero-cli> /build/source/src/rpc/core_rpc_server.cpp:3270:0:   required from here
monero-cli>  3270 |       const std::string span_connection_id = epee::string_tools::pod_to_hex(span.connection_id);
monero-cli> /build/source/contrib/epee/include/span.h:165:34: error: static assertion failed: source type may have padding
monero-cli>   165 |     static_assert(!has_padding<T>(), "source type may have padding");
monero-cli>       |                    ~~~~~~~~~~~~~~^~
monero-cli> /build/source/contrib/epee/include/span.h:165:34: note: '! epee::has_padding<boost::uuids::uuid>()' evaluates to false
monero-cli> make[2]: *** [src/rpc/CMakeFiles/obj_rpc.dir/build.make:104: src/rpc/CMakeFiles/obj_rpc.dir/core_rpc_server.cpp.o] Error 1
monero-cli> make[1]: *** [CMakeFiles/Makefile2:2114: src/rpc/CMakeFiles/obj_rpc.dir/all] Error 2
monero-cli> make: *** [Makefile:146: all] Error 2
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.64, NixOS, 25.05 (Warbler), 25.05.20241212.2f91aa2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(wogan): `""`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/gffj8gl72kb46j9k5g1yg1qc5jrzb7b4-source`

## Notify maintainers

@philipmw @rnhmjoj

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/blockchains/monero-cli/default.nix
+++ b/pkgs/applications/blockchains/monero-cli/default.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   cmake,
   pkg-config,
   boost,
@@ -40,7 +41,6 @@ let
     rev = "bff7fdfe436c727982cc553bdfb29a9021b423b0";
     hash = "sha256-VNypeEz9AV0ts8X3vINwYMOgO8VpNmyUPC4iY3OOuZI=";
   };
-
 in
 stdenv.mkDerivation rec {
   pname = "monero-cli";
@@ -53,7 +53,14 @@ stdenv.mkDerivation rec {
     hash = "sha256-nDiFJjhsISYM8kTgJUaPYL44iyccnz5+Pd5beBh+lsM=";
   };
 
-  patches = [ ./use-system-libraries.patch ];
+  patches = [
+    ./use-system-libraries.patch
+    # https://github.com/monero-project/monero/pull/9462
+    (fetchpatch2 {
+      url = "https://github.com/monero-project/monero/commit/65568d3a884857ce08d1170f5801a6891a5c187c.patch?full_index=1";
+      hash = "sha256-Btuy69y02UyVMmsOiCRPZhM7qW5+FRNujOZjNMRdACQ=";
+    })
+  ];
 
   postPatch = ''
     # manually install submodules
@@ -91,9 +98,7 @@ stdenv.mkDerivation rec {
       libusb1
       protobuf_21
     ]
-    ++ lib.optionals (trezorSupport && stdenv.hostPlatform.isLinux) [
-      udev
-    ];
+    ++ lib.optionals (trezorSupport && stdenv.hostPlatform.isLinux) [ udev ];
 
   cmakeFlags =
     [

